addons/website_slides/models/slide_slide_resource.py

```diff
@@ -48,3 +51,12 @@ def _check_link_type(self):
         for record in self:
             if record.resource_type != 'file' and record.data:
                 raise ValidationError(_("Resource %(resource_name)s is a link and should not contain a data file", resource_name=record.name))
+
+    def _get_download_url(self):
```

Comment:
why not making it a computed field ?

Reply:
I thought "because we usually don't add fields in stable versions unless it's absolutely necessary", and "upgrade will complain even though it's not stored". That said, we could still use this implementation for master in case it is indeed a problem (and forget my comment otherwise).

Reply:
I can't remember where I got this comment in writing, but my understanding was that even unstored fields would be flagged as missing if a db is upgraded and the fix isn't yet merged in the following version.
Surely "Knowledge" will prove handy for this sort of things... Maybe @awa-odoo can confirm or correct me one last time?
Merging in master than backporting is technically an option, but it may not be worth the overhead in this case (vs full forward port than cleanup IMP)

Reply:
look like "template enterprise" is not happy indeed, let's keep the method for 16.0

Reply:
Yes, we discussed that on the task with Pierre-Yves (see chatter).

To quote myself:

> To strictly answer your question, adding fields in stable is not possible for stored fields, and discouraged (but possible) for computed non-stored fields.
However, it should be a last resort solution, as it requires an exception in the runbot and is not great overall.

Adding a field in stable requires bypassing checks and adding exceptions on the runbot. (aka manually asking the runbot team and motivating our changes).
Even if it does not change anything schema-wise, it's still changing the model definition in a stable version, which is not ideal.
If we can avoid it (and if it's not very very hard to avoid it), then we should.

Reply:
Thanks for all your suggestions. I did remember that it was not recommended but not from where, sorry. So I have reverted the change to the method and created another branch in master to change it to a computed field, to rebase and merge when the fix is merged.